File: ruby_small_problems/easy-2/08-sum-or-product.rb
# Write a program that asks the user to enter an integer greater than 0, then 
# asks if the user wants to determine the sum or product of all numbers between
# 1 and the entered integer.

num = ''

loop do
  puts "Please enter a number greater than 0."
  num = gets.chomp.to_i
  break if num > 0
  puts "Something went wrong. Try again."
end

loop do
  puts "Enter 's' to compute the sum, 'p' to compute the product:"
  response = gets.chomp.downcase

  if response == 's'
    puts "The sum is #{(1..num).inject(&:+)}"
    break
  elsif response == 'p'
    puts "The product is #{(1..num).inject(&:*)}"
    break
  else
    puts "Something went wrong. Please enter 's' or 'p'."
  end
end

File: python/small_problems/easy_2/08_sum_or_product.py
# Write a program that asks the user to enter an integer greater than 0,
# then asks if the user wants to determine the sum or product of all
# numbers between 1 and the entered integer.
import math

print("Please enter a positive whole number:")
num = int(input())

print(f"Would you like the (s)um or (p)roduct of all the numbers between 1 and {num}?")

while True:
    choice = input()
    if choice.lower() in ['s', 'p', 'sum', 'product']:
        break
    print(f"I'm sorry, '{choice}' was not a valid input. Please try again")

if choice[0] == 's':
    print(f"{sum(range(1, num + 1))}")
else:
    print(f"{math.prod(range(1, num + 1))}")

